datalad/support/gitrepo.py

```diff
@@ -2362,15 +2370,115 @@ def gc(self, allow_background=False, auto=False):
             cmd_options += ['--auto']
         self._git_custom_command('', cmd_options)
 
-    def get_submodules(self, sorted_=True):
-        """Return a list of git.Submodule instances for all submodules"""
-        # check whether we have anything in the repo. if not go home early
-        if not self.repo.head.is_valid():
-            return []
-        submodules = self.repo.submodules
+    def _parse_gitmodules(self):
+        # TODO read .gitconfig from Git blob?
+        gitmodules = self.pathobj / '.gitmodules'
+        if not gitmodules.exists():
+            return {}
+        # pull out file content
+        out, err = self._git_custom_command(
+            '',
+            ['git', 'config', '-z', '-l', '--file', '.gitmodules'])
+        # abuse our config parser
+        db, _ = _parse_gitconfig_dump(out, {}, None, True)
+        mods = {}
+        for k, v in iteritems(db):
+            if not k.startswith('submodule.'):
+                # we don't know what this is
+                lgr.debug("Skip unrecognized .gitmodule specification: %s=%s", k, v)
+                continue
+            k_l = k.split('.')
+            # module name is everything after 'submodule.' that is not the variable
+            # name
+            mod_name = '.'.join(k_l[1:-1])
+            mod = mods.get(mod_name, {})
+            # variable name is the last 'dot-free' segment in the key
+            mod[k_l[-1]] = v
+            mods[mod_name] = mod
+
+        out = {}
+        # bring into traditional shape
+        for name, props in iteritems(mods):
+            if 'path' not in props:
+                lgr.debug("Failed to get '%s.path', skipping section", name)
+                continue
+            modprops = {'gitmodule_{}'.format(k): v
+                        for k, v in iteritems(props)
+                        if not (k.startswith('__') or k == 'path')}
+            modpath = self.pathobj / PurePosixPath(props['path'])
```

Comment:
I wonder if eventually we should RF this to store just `props['path']` instead of  constructing full path here with intention later on to convert to relative.